src/test/java/org/javarosa/core/model/SelectChoiceTest.java

```diff
@@ -119,4 +123,103 @@ public void getChild_updates_whenChoicesAreFromRepeat() throws IOException {
+            )));
+
+        try {
+            scenario.choicesOf("/data/select").get(0).getChild("invalid-property");
+            fail();
+        } catch (IllegalStateException e) {
+            // expected
+        }
+    }
+
+    @Test
+    public void getAdditionalChildren_returnsChildren_whenChoicesAreFromSecondaryInstance() {
+        setUpSimpleReferenceManager(r("external-select-geojson.xml").getParent(), "file");
+
+        Scenario scenario = Scenario.init("external-select-geojson.xml");
+
+        Map<String, String> firstNodeChildren = scenario.choicesOf("/data/q").get(0).getAdditionalChildren();
+        assertThat(firstNodeChildren.keySet(), hasSize(3));
+        assertThat(firstNodeChildren, hasEntry("geometry", "0.5 102 0 0"));
+        assertThat(firstNodeChildren, hasEntry("id", "fs87b"));
+        assertThat(firstNodeChildren, hasEntry("foo", "bar"));
+
+        Map<String, String> secondNodeChildren = scenario.choicesOf("/data/q").get(1).getAdditionalChildren();
+        assertThat(secondNodeChildren.keySet(), hasSize(4));
+        assertThat(secondNodeChildren, hasEntry("geometry", "0.5 104 0 0"));
+        assertThat(secondNodeChildren, hasEntry("id", "67abie"));
+        assertThat(secondNodeChildren, hasEntry("foo", "quux"));
+        assertThat(secondNodeChildren, hasEntry("special-property", "special value"));
+    }
+
+    @Test
+    public void getChildren_updates_whenChoicesAreFromRepeat() throws IOException {
+        Scenario scenario = Scenario.init("Select from repeat", html(
+            head(
+                title("Select from repeat"),
+                model(
+                    mainInstance(
+                        t("data id='repeat-select'",
+                            t("repeat",
+                                t("value"),
+                                t("label"),
+                                t("special-property")),
+                            t("filter"),
+                            t("select"))))),
+            body(
+                repeat("/data/repeat",
+                    input("value"),
+                    input("label"),
+                    input("special-property")),
+                input("filter"),
+                select1Dynamic("/data/select", "../repeat")
+            )));
+        scenario.answer("/data/repeat[0]/value", "a");
+        scenario.answer("/data/repeat[0]/label", "A");
+        scenario.answer("/data/repeat[0]/special-property", "AA");
+
+        assertThat(scenario.choicesOf("/data/select").get(0).getValue(), is("a"));
+        Map<String, String> children = scenario.choicesOf("/data/select").get(0).getAdditionalChildren();
+        assertThat(children.keySet(), hasSize(2));
+        assertThat(children, hasEntry("value", "a"));
+        assertThat(children, hasEntry("special-property", "AA"));
+
+        scenario.answer("/data/repeat[0]/special-property", "changed");
+        children = scenario.choicesOf("/data/select").get(0).getAdditionalChildren();
```

Comment:
I hadn't really thought about this previously -- because `getChild` and `getAdditionalChildren` use the display text of the children, they're not dynamic. That is, you have to call `getChild` and `getAdditionalChildren` again after the form changes to get new values. I think that's ok -- I can't think of a usage where we'd hold on to one of these while other parts of the form are changing but I still wanted to flag it.

The alternative would be to keep a reference to an `AnswerData` object instead. That would reflect changes to the model.